### ISSUE TITLE
Also parse parent class from 3rd party Jar files

### ISSUE THREAD

**Issue by pascalgn** [havardh/javaflow#68]
Let's say we have the following java types:

```java
public abstract class BaseDto {
    private int id;

    public int getId() {
        return id;
    }
}

public class UserDto extends BaseDto {
    private String name;

    public String getName() {
        return name;
    }
}
```

then `javaflow` will correctly generate

```js
export UserDto = {
  id: number,
  name: string
};
```

_However_, when `BaseDto` is not from the same project, but actually only included via Jar file, `javaflow` will not be able to parse BaseDto (source code is not available) and the resulting type UserDto will be missing the `id` field.

**Comment by havardh** [comment]
I wrote `javaflow` for a quite opinionated project, as you might have noticed from all the missing features you've needed to add. 

We had the rule that the API layer should be completely separated and without external depencies, which lead to `javaflow` only having to look into a single root directory to generate types for a complete API. The few cases where we did depend on external classes (while migrating) we used the the `any` flowtype through custom type substitution. The case above would _work_ as the `flowtype` produced are not [Exact types](https://flow.org/en/docs/types/objects/#toc-exact-object-types).

This opinion is a bit silly to enforce on any project wanting to use `javaflow`. However, including this feature would add significant complexity to the project. I am *not* against adding this at all, but I would think it might require a lot of work to be "complete". My concern is the integration with the applications class loaders. And I have no idea how one would start to facilitate integrations with any of options one might meet. 

So lets try to define this in an incremental fashion, so that we start with the support required to get your project to work. 
I do believe we could add this as a new `Resolver` like the `FileResolver` that we wrote last time. If we could write a `DependencyResolver` which would, given a path to a Java class return a `List<Path>` to any of these "missing" dependencies, using the ClassLoader of your project or just be the default ClassLoader. My intent is that we should be able to apply both the `FileResolver` and the `DependencyResolver` in the same project. 

What do you think? Am I making this too complicated?

**Comment by pascalgn** [comment]
Yes, that was also my first impression: It will require a lot of work. One problem I see is that during a Maven build, I think the class loader only contains the Maven plugin classes (Mojo, Maven classes, etc.), but not the classes of the project's dependencies.

After I read your comment, I got another idea, though: We already allow the user to specify custom types via `types.yml` (or via config in Maven), so we could maybe simplify the problem like this:
1. Parent type is also a known source file? OK, use it
2. Parent type is not known, but a custom type with the same name has been passed? OK, use it
3. Else: Ignore the parent type (as we do now)

So in my example, `types.yml` could look like this:
```yaml
com.example.BaseDto: "{ id: number }"
```

Flow seems to support [Object type spread](https://github.com/facebook/flow/releases/tag/v0.42.0) for some time now, so we would not need to parse the given user types. The generated file could just be

```js
// Types from user's types.yml:
export type BaseDto = { id: number };

export type UserDto = {
  ...BaseDto,
  name: string
};
```

For now, this would be enough for us as we only have a few base types that we would need to pass via `types.yml`. I think this would be a lot less work than having to implement additional class parsing. WDYT?

**Comment by havardh** [comment]
Perfect :)

We could also make more use of yaml and specify it as:

```
com.example.BaseDto:
  id: number
```

That way we would not need to rely on the object spread, while skipping parsing. It would also allow this format to be reused if we target Typescript or Reason down the line :v:

**Comment by havardh** [comment]
I'm not sure about mixing two formats in one file. We will always be able to specify some XML markup that will make those acceptable. It will be clunky and more work to parsed, but I think it will be worth it.
